packages/waku/src/router/define-router.ts

```diff
@@ -368,17 +374,41 @@ export function new_defineRouter(fns: {
           has404,
         };
   };
+  const canSkip = async (
+    pathname: string,
+    slotId: SlotId,
+  ): Promise<boolean> => {
+    const pathConfig = await getMyPathConfig();
+    const found = pathConfig.find(({ pathname: pathSpec }) =>
+      getPathMapping(pathSpec, pathname),
+    );
+    return !!found && found.staticElementIds.includes(slotId);
+  };
   const renderEntries: RenderEntries = async (rscPath, { rscParams }) => {
     const pathname = decodeRoutePath(rscPath);
     const pathStatus = await existsPath(pathname);
     if (!pathStatus.found) {
       return null;
     }
     const { query, skip } = parseRscParams(rscParams);
-    const entries = await fns.renderRoute(
+    const { routeElement, elements } = await fns.renderRoute(
       pathname,
-      pathStatus.isStatic ? {} : { query, skip },
+      pathStatus.isStatic ? {} : { query },
     );
+    if (
+      Object.keys(elements).some((id) => id.startsWith(ROUTE_SLOT_ID_PREFIX))
+    ) {
+      throw new Error('Element ID cannot start with "route:"');
+    }
```

Comment:
Is this check needed if you're moving to use `routeElement` also?

Reply:
This is needed to avoid conflict because we move to `routeElement`.